test/performance/stress-test.janet

```diff
@@ -0,0 +1,76 @@
+# run with 'time janet stress-test.janet'
+# NOTE: the macros here and incredibly unhygenic with probably a lot of accidental binding bapture!
+
+(use /junk-drawer)
+
+(var ALPHABET ["a" "b" "c" "d" "e" "f" "g" "h" "i"
+               "j" "k" "l" "m" "n" "o" "p" "q" "r"
+               "s" "t" "u" "v" "w" "x" "y" "z"])
+
+(defmacro def-component-alphabet []
+  "Define a component for every letter of the alphabet."
+  ~(upscope
+    ,;(map |(tuple ~def-component (symbol $) :val :number)
+           ALPHABET)))
+
+(defmacro def-systems-alphabet []
+  "Define and register a system with 2 queries, each with 2 alphabet components."
+  ~(upscope
+    # define system
+    ,;(map (fn [i]
+             (tuple ~def-system (symbol "sys-" i)
```

Comment:
Single symbols are better `quote`d with `'`. There is no need to `quasiquote` them with `~`.